packages/typespec-vscode/src/emit/emit.ts

```diff
@@ -0,0 +1,335 @@
+      package: e.package,
+      emitterKind: e.kind,
+      label: e.language,
+      detail: `Generate ${e.language} ${e.kind} code from ${e.package}`,
+      picked: false,
+      fromConfig: false,
+      iconPath: Uri.file(context.asAbsolutePath(`./icons/${e.language.toLowerCase()}.svg`)),
+    };
+  };
+
+  const registerEmitters = getRegisterEmitters(kind);
+  const all = [...registerEmitters].map((e) => toQuickPickItem(e));
+
+  const selectedEmitter = await vscode.window.showQuickPick<EmitQuickPickItem>(all, {
+    title: "Select a Language",
+    canPickMany: false,
+    placeHolder: "Pick a Language",
+    ignoreFocusOut: true,
+  });
+
+  if (!selectedEmitter) {
+    logger.info("No emitter selected. Generating Cancelled.", [], {
+      showOutput: false,
+      showPopup: false,
+    });
+    return;
+  }
+
+  logger.info("npm install...", [], {
+    showOutput: false,
+    showPopup: false,
+  });
+
+  const npmUtil = new NpmUtil(baseDir);
+  const packagesToInstall: string[] = [];
+
+  /* install emitter package. */
+  logger.info(`select ${selectedEmitter.package}`);
+  const { action, version } = await npmUtil.ensureNpmPackageInstallAction(
+    selectedEmitter.package,
+    selectedEmitter.version,
+  );
+
+  if (action === InstallationAction.Upgrade) {
+    logger.info(`Upgrading ${selectedEmitter.package} to version ${version}`);
+    const options = {
+      ok: `OK (install ${selectedEmitter.package}@${version} by 'npm install'`,
+      recheck: `Check again (install ${selectedEmitter.package} manually)`,
+      ignore: `Ignore (don't upgrade emitter ${selectedEmitter.package})`,
+    };
+    const selected = await vscode.window.showQuickPick(Object.values(options), {
+      canPickMany: false,
+      ignoreFocusOut: true,
+      placeHolder: `Package '${selectedEmitter.package}' needs to be upgraded for generating`,
+      title: `TypeSpec Generating...`,
+    });
+    if (selected === options.ok) {
+      packagesToInstall.push(`${selectedEmitter.package}@${version}`);
+    } else if (selected === options.ignore) {
+      logger.info(`Ignore upgrading emitter ${selectedEmitter.package} for generating`, [], {
+        showOutput: false,
+        showPopup: false,
+      });
+    } else {
+      logger.info(
+        `Need to manually install the package ${selectedEmitter.package}@${version}. Generating Cancelled.`,
+        [],
+        {
+          showOutput: false,
+          showPopup: true,
+        },
+      );
+      return;
+    }
+  } else if (action === InstallationAction.Install) {
+    let packageFullName = selectedEmitter.package;
+    if (version) {
+      packageFullName = `${selectedEmitter.package}@${version}`;
+    }
+    logger.info(`Installing ${packageFullName}`);
+    /* verify dependency packages. */
+    const dependenciesToInstall = await npmUtil.ensureNpmPackageDependencyToUpgrade(
+      selectedEmitter.package,
+      version,
+      npmDependencyType.peerDependencies,
```

Comment:
dont we need to check dep and devdep?

Reply:
Current all the emitters has peer dependency, current we only need check peerDependencies. In future, when we support customized emitter which may not follow our rule to define peerDependencies, we will add more dependency check.

Reply:
but we already support customized emitters through config, right?..